dhcpv4/bsdp/bsdp.go

```diff
@@ -0,0 +1,315 @@
+	// If less than length of boot image ID and count, it's probably invalid.
+	if len(bytes) < 5 {
+		return nil, fmt.Errorf("not enough bytes for BootImageID")
+	}
+	imageID := BootImageIDFromBytes(bytes[:4])
+	nameLength := int(bytes[4])
+	if 5+nameLength > len(bytes) {
+		return nil, fmt.Errorf("not enough bytes for BootImage")
+	}
+	name := string(bytes[5 : 5+nameLength])
+	return &BootImage{ID: imageID, Name: name}, nil
+}
+
+// makeVendorClassIdentifier calls the sysctl syscall on macOS to get the
+// platform model.
+func makeVendorClassIdentifier() (string, error) {
+	// Fetch hardware model for class ID.
+	hwModel, err := syscall.Sysctl("hw.model")
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("AAPLBSDPC/i386/%s", hwModel), nil
+}
+
+// ParseBootImagesFromOption parses data from the BSDPOptionBootImageList
+// option and returns a list of BootImages.
+func ParseBootImagesFromOption(data []byte) ([]BootImage, error) {
+	// Should at least have the # bytes of boot images.
+	if len(data) < 4 {
+		return nil, fmt.Errorf("invalid length boot image list")
+	}
+
+	var (
+		readByteCount = 0
+		start         = data
+		bootImages    []BootImage
+	)
+	for {
+		bootImage, err := BootImageFromBytes(start)
+		if err != nil {
+			return nil, err
+		}
+		bootImages = append(bootImages, *bootImage)
+		// Read BootImageID + name length + name
+		readByteCount += 4 + 1 + len(bootImage.Name)
+		if readByteCount+1 >= len(data) {
+			break
+		}
+		start = start[readByteCount:]
+	}
+
+	return bootImages, nil
+}
+
+// ParseVendorOptionsFromOptions extracts the sub-options list of the vendor-
+// specific options from the larger DHCP options list.
+func ParseVendorOptionsFromOptions(options []dhcpv4.Option) []dhcpv4.Option {
+	var (
+		vendorOpts []dhcpv4.Option
+		err        error
+	)
+	for _, opt := range options {
+		if opt.Code == dhcpv4.OptionVendorSpecificInformation {
+			vendorOpts, err = dhcpv4.OptionsFromBytesWithoutMagicCookie(opt.Data)
+			if err != nil {
+				return []dhcpv4.Option{}
+			}
+			break
+		}
+	}
+	return vendorOpts
+}
+
+// ParseBootImageListFromAck parses the list of boot images presented in the
+// ACK[LIST] packet and returns them as a list of BootImages.
+func ParseBootImageListFromAck(ack dhcpv4.DHCPv4) ([]BootImage, error) {
+	var bootImages []BootImage
+	vendorOpts := ParseVendorOptionsFromOptions(ack.Options())
+	for _, opt := range vendorOpts {
+		if opt.Code == OptionBootImageList {
+			images, err := ParseBootImagesFromOption(opt.Data)
+			if err != nil {
+				return nil, err
+			}
+			bootImages = append(bootImages, images...)
+		}
+	}
+
+	return bootImages, nil
+}
+
+// NewInformListForInterface creates a new INFORM packet for interface ifname
+// with configuration options specified by config.
+func NewInformListForInterface(iface string, replyPort uint16) (*dhcpv4.DHCPv4, error) {
+	d, err := dhcpv4.NewInformForInterface(iface /* needsBroadcast = */, false)
```

Comment:
the comma should be before the commented flag:

```
d, err := dhcpv4.NewInformForInterface(iface, /* needsBroadcast = */ false)
```

Reply:
Weird I thought I fixed that, must have gotten lost in rebasing.

Reply:
Huh, goimports/gofmt seems to prefer the other way...

Reply:
What error do you get? I just tried with a simple program and golint didn't complain

```Go
package main

func blah(a string, b bool) {
}

func main() {
        blah("test", /* b = */ true)
}
```

Reply:
Not sure how much I can do on this one.

Reply:
just ignore it for now. Out of curiousity I'll have a look at it once merged